apps/chat/src/components/Chat/Migration/MigrationFailedModal.tsx

```diff
@@ -0,0 +1,315 @@
+  withPt?: boolean;
+}
+
+const ItemsList = <T extends Conversation | Prompt>({
+  entitiesToRetryIds,
+  setEntitiesToRetryIds,
+  failedMigratedEntities,
+  getModelIcon,
+  withPt,
+}: ItemsListProps<T>) => {
+  const handleSelect = (entityId: string) => {
+    setEntitiesToRetryIds([...new Set([...entitiesToRetryIds, entityId])]);
+  };
+
+  const handleUnselect = (entityId: string) => {
+    setEntitiesToRetryIds(entitiesToRetryIds.filter((id) => id !== entityId));
+  };
+
+  return (
+    !!failedMigratedEntities.length && (
+      <div className={classNames('mt-2', withPt && 'pt-2')}>
+        <ul className="flex flex-col gap-0.5">
+          {failedMigratedEntities.map((entity) => (
+            <li
+              className="relative flex h-[30px] items-center justify-between rounded"
+              key={entity.id}
+            >
+              <div className="flex">
+                {getModelIcon(entity)}
+                <p className="ml-2">{entity.name}</p>
+              </div>
+              <div className="flex w-[100px] items-center justify-around">
+                <div
+                  onClick={() => handleSelect(entity.id)}
+                  className="relative flex size-[18px] group-hover/file-item:flex"
+                >
+                  <input
+                    className="checkbox peer size-[18px] bg-transparent"
+                    type="checkbox"
+                    readOnly
+                    checked={entitiesToRetryIds.includes(entity.id)}
+                  />
+                  <IconCheck
+                    size={18}
+                    className="pointer-events-none invisible absolute text-accent-primary peer-checked:visible"
+                  />
+                </div>
+                <div
+                  onClick={() => handleUnselect(entity.id)}
+                  className="relative flex size-[18px] group-hover/file-item:flex"
+                >
+                  <input
+                    className="checkbox peer size-[18px] bg-transparent"
+                    type="checkbox"
+                    readOnly
+                    checked={!entitiesToRetryIds.includes(entity.id)}
+                  />
+                  <IconCheck
+                    size={18}
+                    className="pointer-events-none invisible absolute text-accent-primary peer-checked:visible"
+                  />
+                </div>
+              </div>
+            </li>
+          ))}
+        </ul>
+      </div>
+    )
+  );
+};
+
+interface Props {
+  failedMigratedConversations: Conversation[];
+  failedMigratedPrompts: Prompt[];
+}
+
+export const MigrationFailedWindow = ({
+  failedMigratedConversations,
+  failedMigratedPrompts,
+}: Props) => {
+  const { t } = useTranslation(Translation.Common);
+
+  const dispatch = useDispatch();
+
+  const router = useRouter();
+
+  const [conversationsToRetryIds, setConversationsToRetryIds] = useState(() =>
+    failedMigratedConversations.map((conv) => conv.id),
```

Comment:
JFYI - if failedMigratedConversations will be updated after this useState will done you will have incorrect values
To fix it you should have the same set in useEffect